misc.Rmd

```diff
@@ -65,69 +80,45 @@ citation()
 
-A demo is an `.R` file that lives in `demo/`.
-Demos are like examples but tend to be longer.
-Instead of focussing on a single function, they show how to weave together multiple functions to solve a problem.
+We bring this up because, in practice, some packages use the `tools/` directory for a different but related purpose.
+Some packages have periodic maintenance tasks for which it is helpful to record detailed instructions.
+For example, many packages embed some sort of external resource, e.g. code or data:
 
-You list and access demos with `demo()`:
+-   Source code and headers for an embedded third party C/C++ library
 
--   Show all available demos: `demo()`.
--   Show all demos in a package: `demo(package = "httr")`.
--   Run a specific demo: `demo("oauth1-twitter", package = "httr")`.
--   Find a demo: `system.file("demo", "oauth1-twitter.R", package = "httr")`.
+-   Web toolkits
 
-Each demo must be listed in `demo/00Index` in the following form: `demo-name   Demo description`.
-The demo name is the name of the file without the extension, e.g. `demo/my-demo.R` becomes `my-demo`.
+-   R code that's inlined (as opposed to imported)
 
-By default the demo asks for human input for each plot: "Hit <Return> to see next plot:".
-This behaviour can be overridden by adding `devAskNewPage(ask = FALSE)` to the demo file.
-You can add pauses by adding: `readline("press any key to continue")`.
+-   Specification for a web API
 
-Generally, I do not recommend using demos.
-Instead, consider writing a vignette:
+-   Colour palettes, styles, and themes
 
--   Demos are not automatically tested by `R CMD check`.
-    This means that they can easily break without your knowledge.
+-   Database of time zones
 
--   Vignettes have both input and output, so readers can see the results without having to run the code themselves.
+These external assets are also usually evolving over time, so they need to be re-ingested on a regular basis.
+This makes it particularly rewarding to implement such housekeeping programmatically.
 
--   Longer demos need to mingle code with explanation, and RMarkdown is better suited to that task than R comments.
+So this is a second, unofficial use of the `tools/` directory, characterized by two big differences with its official purpose: The packages that do this generally do not have a `configure` script and they list `tools/` in `.Rbuildignore`, meaning that these scripts are not included in the package bundle.
+These scripts are maintained in the source package for developer convenience, but are never shipped with the package.
 
--   Vignettes are listed on the CRAN package page.
-    This makes it easier for new users to discover them.
+This practice is closely related to our recommendation to of store the instructions for the creation of package data in `data-raw/` (section @sec-data-data-raw) and to record the method of construction for any test fixtures (section @sec-testing-advanced-concrete-fixture).
```

Comment:
I still want to make this connection, even if I'm not pursuing it any further at this time. But I think it's interesting to think about generalizing `data-raw/`. We had already talked about the connection with creating concrete test fixtures.